ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -33,9 +78,17 @@ cpdef traceback_to_frames(traceback, max_nframes):
     while tb is not None:
         if nframes < max_nframes:
             frame = tb.tb_frame
-            code = frame.f_code
-            lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.insert(0, (code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
+            # Python 3.11 moved PyFrameObject members to internal C API and cannot be directly accessed
+            IF PY_MAJOR_VERSION > 3 or (PY_MAJOR_VERSION == 3 and PY_MINOR_VERSION >= 11):
+                code = PyFrame_GetCode(<PyFrameObject*> frame)
+                lineno = PyFrame_GetLineNumber(<PyFrameObject*> frame)
+                lineno = 0 if lineno is None else lineno
+                frames.insert(0, ((<object>code).co_filename, lineno, (<object>code).co_name, _extract_class_name(frame)))
```

Comment:
should you cast `code` to `object` once? does it matter?

could `code` ever be `NULL` ?

Reply:
What happens if we cast the return value of `PyFrame_GetCode` to `<object>` instead?